compiler/src/dotty/tools/dotc/core/DenotTransformers.scala

```diff
@@ -24,9 +24,9 @@ object DenotTransformers {
     /** The last phase during which the transformed denotations are valid */
     def lastPhaseId(implicit ctx: Context) = ctx.nextDenotTransformerId(id + 1)
 
-    /** The validity period of the transformer in the given context */
+    /** The validity period of the transformed denotations in the given context */
     def validFor(implicit ctx: Context): Period =
-      Period(ctx.runId, id, lastPhaseId)
+      Period(ctx.runId, id + 1, lastPhaseId)
```

Comment:
At first I was scared, because it looks really hard to see all the ramifications of this change. But then I found out that the method is in fact not used at all! So do we still want to keep it?